### PR TITLE
Extract CdpTarget from BrowsingContext

### PR DESCRIPTION
**Pull request by sadym-chromium**
The change is needed to ease the enabling domains on-demand. It is required for network domain.

### CODE REVIEW

File: src/bidiMapper/domains/context/cdpTarget.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+export class CdpTarget {
+  readonly #targetUnblocked: Deferred<void>;
+  readonly #targetId: string;
+  readonly #cdpClient: CdpClient;
+  readonly #eventManager: IEventManager;
+  readonly #cdpSessionId: string;
+  readonly #realmStorage: RealmStorage;
+  readonly #browsingContextStorage: BrowsingContextStorage;
+  readonly #logger: LoggerFn | undefined;
```
Comment (thiagowfx) [inline src/bidiMapper/domains/context/cdpTarget.ts:39]:
```suggestion
  readonly #logger?: LoggerFn;
```

File: src/bidiMapper/domains/context/cdpTarget.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+  get cdpClient(): CdpClient {
+    return this.#cdpClient;
+  }
+
+  get cdpSessionId(): string {
+    return this.#cdpSessionId;
+  }
+
+  async unblock() {
```
Comment (thiagowfx) [inline src/bidiMapper/domains/context/cdpTarget.ts:81]: Could you add JSDoc to this method?

File: src/bidiMapper/domains/context/cdpTarget.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
+/*
```
Comment (thiagowfx) [inline src/bidiMapper/domains/context/cdpTarget.ts:1]: Does it make sense to add unit tests to this class?

Comment (sadym-chromium) [reply src/bidiMapper/domains/context/cdpTarget.ts:1]: This class is tightened to the CDP, which makes it extremely hard to unit test. We can rely on e2e tests.

File: src/bidiMapper/domains/context/cdpTarget.ts (hunk 2 above)
Comment (sadym-chromium) [reply src/bidiMapper/domains/context/cdpTarget.ts:81]: done

File: package.json [hunk 4]
```diff
@@ -11,7 +11,7 @@
     "e2e-only": "python3 -m pytest",
     "eslint": "eslint --ext .js,.ts --fix .",
     "flake8": "flake8 examples/ tests/",
-    "format": "npm run eslint && npm run prettier & npm run yapf",
+    "format": "npm run eslint && npm run prettier && npm run yapf",
```
Comment (sadym-chromium) [inline package.json:14]: otherwise it hangs

Comment (thiagowfx) [reply package.json:14]: OK

File: .prettierignore [hunk 5]
```diff
@@ -1,5 +1,7 @@
 examples/*.html
 lib/
+logs/
 src/.build
 tests/__pycache__
 wpt/
+wpt-metadata/*/
```
Comment (thiagowfx) [inline .prettierignore:7]:
```suggestion
wpt-metadata/
```

**Review by thiagowfx — APPROVED** [review]

File: .prettierignore (hunk 5 above)
Comment (sadym-chromium) [reply .prettierignore:7]: I still want the readme to be formatted

**Pull request merged**